skrub/_join_utils.py

```diff
@@ -84,12 +84,15 @@ def check_missing_columns(table, key, table_name):
     table : DataFrame
         The table that should contain the columns listed in `key`.
 
-    key : list of str
+    key : list of str or selector
         List of column names, all of which must be found in `table`.
 
     table_name : str
         Name by which to refer to `table` in the error message if necessary.
     """
+    if isinstance(key, s.Selector):
```

Comment:
This function could be simplified with:
```python
def check_missing_columns(table, key):
    s.make_selector(key).expand(table)
    return
```
or
```python
def check_missing_columns(table, key, table_name):
    s.make_selector(key).expand(table, table_name)
    return
```
@jeromedockes I think adding a `table_name` argument to the `expand` function error message would be a nice addition for debuggability, WDYT?

Reply:
that sounds useful. another way would be to have Cols raise a custom exception with an attribute listing the missing columns, and then whoever is calling it and catching the exception can retreive that and format it however they see fit in their error message. It would allow less generic error messages and avoid adding the table name argument that only one selector type needs

Reply:
Yes, that looks even better

Reply:
I think this is a good idea, and that it deserves its own PR -- I wanted to only change `AggJoiner` / `AggTarget` here